### ISSUE TITLE
Empty array becomes null when synced to Firebase

### ISSUE THREAD

**Issue by huntercaron** [huntercaron/firebase-data#2]
```js
const data = firebaseData({
    messages: []
})

function addToArray() {
    // this will error due to array not existing
    data.messages.push("Hey bob")
}

addToArray()
```

**Comment by huntercaron** [comment]
Would it be wild to create a string constant that I sub in for empty arrays? @pburtchaell

**Comment by pburtchaell** [comment]
That could work. Thought about this the other day and the string constant came to mind as well—maybe something obscure, yet descriptive like `$$firebaseNullArray` or `$$firebase_null_array`?

Also, found this—worth a read: 
https://firebase.googleblog.com/2014/04/best-practices-arrays-in-firebase.html

**Comment by pburtchaell** [comment]
Actually, `$$firebaseEmptyArray` might be more descriptive? Null array is a bit weird, because a value technically can't be both null and an array. What do you think?

**Comment by pburtchaell** [comment]
Also the reason for the `$$` naming pattern:
https://overreacted.io/why-do-react-elements-have-typeof-property/

**Comment by huntercaron** [comment]
Good call, I made an attempt with that string name in #5

**Comment by huntercaron** [comment]
Fixed in #5

**Comment by huntercaron** [comment]
Gonna attempt this again with JSON.stringify()
